zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.zookeeper;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.zookeeper.ZooKeeperException;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerListener;
+
+/**
+ * A ZooKeeper Server Listener.When you add this listener, server will be automatically registered
+ * into the ZooKeeper.
+ */
+public class ZooKeeperUpdatingListener implements ServerListener {
+    private final String zkConnectionStr;
+    private final String zNodePath;
+    private final int sessionTimeout;
+    private Endpoint endpoint;
```

Comment:
@jonefeewang OK. Thanks for explanation. I think we're good to go once you address the remaining comments.